couchdb/couchdb.go

```diff
@@ -154,12 +166,57 @@ func DeleteDB(dbprefix, doctype string) error {
 }
 
 // ResetDB destroy and recreate the database for a doctype
-func ResetDB(dbprefix, doctype string) error {
-	err := DeleteDB(dbprefix, doctype)
-	if err != nil {
-		return err
+func ResetDB(dbprefix, doctype string) (err error) {
+	err = DeleteDB(dbprefix, doctype)
+	if err == nil {
+		return CreateDB(dbprefix, doctype)
+	}
+	return
```

Comment:
The goal is to make the code easier to change if we had to add a third step, like this:

```go
func ResetDB(dbprefix, doctype string) error {
  err := DeleteDB(dbprefix, doctype)
  if err != nil {
    return err
  }
  err = CreateDB(dbprefix, doctype)
  if err != nil {
    return err
  }
  return ThirdStep()
}
```